src/components/shared/PieChartContainer.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Legend, PieChart, Pie, Sector, ResponsiveContainer } from 'recharts';
+
+const renderActiveShape = props => {
+  const RADIAN = Math.PI / 180;
+  const {
+    cx,
+    cy,
+    midAngle,
+    innerRadius,
+    outerRadius,
+    startAngle,
+    endAngle,
+    fill,
+    percent,
+    value,
+    name,
+  } = props;
+  const sin = Math.sin(-RADIAN * midAngle);
+  const cos = Math.cos(-RADIAN * midAngle);
+  const sx = cx + (outerRadius + 10) * cos;
+  const sy = cy + (outerRadius + 10) * sin;
+  const mx = cx + (outerRadius + 30) * cos;
+  const my = cy + (outerRadius + 30) * sin;
+  const ex = mx + (cos >= 0 ? 1 : -1) * 22;
+  const ey = my;
+  const textAnchor = cos >= 0 ? 'start' : 'end';
+
+  return (
+    <g>
+      <text x={cx} y={cy} dy={8} textAnchor="middle" fill={fill}>
+        {name}
+      </text>
+      <Sector
+        cx={cx}
+        cy={cy}
+        innerRadius={innerRadius}
+        outerRadius={outerRadius}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        fill={fill}
+      />
+      <Sector
+        cx={cx}
+        cy={cy}
+        startAngle={startAngle}
+        endAngle={endAngle}
+        innerRadius={outerRadius + 6}
+        outerRadius={outerRadius + 10}
+        fill={fill}
+      />
+      <path
+        d={`M${sx},${sy}L${mx},${my}L${ex},${ey}`}
+        stroke={fill}
+        fill="none"
+      />
+      <circle cx={ex} cy={ey} r={2} fill={fill} stroke="none" />
+      <text
+        x={ex + (cos >= 0 ? 1 : -1) * 12}
+        y={ey}
+        textAnchor={textAnchor}
+        fill="#333"
+      >{`${name}: ${value}`}</text>
+      <text
+        x={ex + (cos >= 0 ? 1 : -1) * 12}
+        y={ey}
+        dy={18}
+        textAnchor={textAnchor}
+        fill="#999"
+      >
+        {`(${(percent * 100).toFixed(2)}%)`}
+      </text>
+    </g>
+  );
+};
+
+renderActiveShape.propTypes = {
+  cx: PropTypes.number,
+  cy: PropTypes.number,
+  midAngle: PropTypes.number,
+  innerRadius: PropTypes.number,
+  outerRadius: PropTypes.number,
+  startAngle: PropTypes.number,
+  endAngle: PropTypes.number,
+  fill: PropTypes.string,
+  percent: PropTypes.number,
+  value: PropTypes.number,
+  name: PropTypes.string,
+};
+class PieChartContainer extends Component {
+  static propTypes = {
+    cellData: PropTypes.array,
+    data: PropTypes.array,
+  };
+
+  state = {
+    activePieIndex: 0,
+  };
+
+  onPieEnter = (cell, index) => {
+    this.setState({
+      activePieIndex: index,
+    });
+  };
+
+  render() {
+    const { activePieIndex } = this.state;
+    const { cellData, data } = this.props;
```

Comment:
Keep some extra props like `showLegend` since it is a shared component, so make it customizable.

Reply:
@harsh-jindal This comment isn't addressed in the PR.

Reply:
@akshatnitd I didn't get what you mean by showLegend ?

Reply:
Keep a boolean prop for this component, `showLegend`, by which we can show/hide the Legend of the pie-chart.

Reply:
@harsh-jindal Please finalise the PR

Reply:
@akshatnitd Done